torch_disc/torch_disc/csrc/disc_compiler/passes/cluster.cpp

```diff
@@ -24,40 +24,63 @@ using namespace ::torch::jit;
 // conversion module into a group. We should re-implement this.
 std::vector<Node*> FakeCluster(const std::shared_ptr<Graph>& graph) {
   std::vector<Node*> nodes;
-  for (auto node : graph->nodes()) {
+  auto is_disc_compilable = [](torch::jit::Node* node) {
     if (torch::blade::IsMlirMhloSupported(*node) &&
         node->kind() != prim::Constant) {
-      if (node->kind() == aten::addmm) continue;
-      nodes.push_back(node);
+      for (auto& input : node->inputs()) {
+        // input should be Tensor or Scalar with explict type
+        auto typ = input->type();
+        if (!typ->cast<c10::TensorType>() &&
+            c10::tryScalarTypeFromJitType(*typ) == c10::nullopt)
+          return false;
+      }
+      return true;
     }
-  }
+    return false;
+  };
+
+  std::copy_if(graph->nodes().begin(), graph->nodes().end(),
+               std::back_inserter(nodes), is_disc_compilable);
   return nodes;
 }
 
+c10::TypePtr getScalarTypePtr(at::ScalarType& typ) {
```

Comment:
Please check if there is such a mapping somewhere in our project already. cc @fortianyou